__tests__/app.test.js

```diff
@@ -168,33 +189,167 @@ describe("/api/articles/:article_id/comments", () => {
+  return request(app)
+    .post("/api/articles/1/comments")
+    .send({
+      username: "not_a_user",
+      body: "This comment will not be accepted.",
+    })
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Username not found");
+    });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("PATCH: 200 responds with the updated article", () => {
+    // Check how many votes there are to expect 1 more
+    return request(app)
+      .get("/api/articles/1")
+      .then((res) => {
+        const initialVotes = res.body.article.votes;
+        return request(app)
+          .patch("/api/articles/1")
+          .send({ upd_votes: 1 })
+          .expect(200)
+          .then((response) => {
+            expect(response.body.article.votes).toBe(initialVotes + 1);
+          });
+      });
+  });
+
+  test("PATCH: 400 responds with an error message for invalid 'upd_votes'", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ upd_votes: "string" })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "Invalid input: 'upd_votes' must be a number"
+        );
+      });
+  });
+
+  test("PATCH: 400 responds with an error message for invalid article_id", () => {
+    return request(app)
+      .patch("/api/articles/invalidID")
+      .send({ upd_votes: 1 })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid input");
+      });
+  });
+
+  test("PATCH: 404 responds with an error message for non-existent article_id", () => {
+    return request(app)
+      .patch("/api/articles/999999")
+      .send({ upd_votes: 1 })
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toBe("Article not found for id: 999999");
+        expect(response.body.msg).toBe("Article Not Found");
       });
   });
 
-  test("POST: 400 responds with an error message for invalid article_id", () => {
+  test("PATCH: 400 responds with an error message for missing 'upd_votes'", () => {
     return request(app)
-      .post("/api/articles/invalidID/comments")
-      .send({ username: "butter_bridge", body: "Great article!" })
+      .patch("/api/articles/1")
+      .send({})
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "Invalid input: 'upd_votes' must be a number"
+        );
+      });
+  });
+});
+
+describe("/api/comments/:comment_id", () => {
+  test("DELETE: 204 responds with no content and successfully deletes a comment", () => {
+    return request(app)
+      .delete("/api/comments/1")
+      .expect(204)
+      .then(() => {
+        return request(app)
+          .get("/api/articles/1/comments")
+          .expect(200)
+          .then((response) => {
+            const comments = response.body.comments;
+            expect(comments).not.toContainEqual(
```

Comment:
Normally with a delete request just checking that a 204 status is given in response is enough of an assertion, but if you still want to return comments for your response then this assertion is cool 😎